### PR TITLE
Adds frontend support for searching inside published dashboards.

### PR DESCRIPTION
**Pull request by kharbutli-meta**
## Description

GTT-1557: Adds frontend support for searching inside of published dashboards. Given a query, searches text fields inside published dashboards, and displays those dashboards with content matching the query. Text snippets are also displayed.

## Testing

 - Added unit tests and tested with it.
 - Tested by running locally.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/src/containers/Home.tsx [hunk 1]
```diff
@@ -11,32 +11,19 @@ import MarkdownRender from "../components/MarkdownRender";
...
 
 function Home() {
-  const [filter, setFilter] = useState("");
   const { homepage, loading } = usePublicHomepage();
   const { t } = useTranslation();
   const dateFormatter = useDateTimeFormatter();
 
   const onSearch = (query: string) => {
-    setFilter(query);
+    window.location.assign("/search/" + query);
```
Comment (miguelpdiaz8) [inline frontend/src/containers/Home.tsx]: You should use the history hook to do this. Look at the AddChart file for reference.

File: frontend/src/containers/Home.tsx [hunk 2]
```diff
@@ -11,32 +11,19 @@ import MarkdownRender from "../components/MarkdownRender";
...
   };
 
   const onClear = () => {
-    setFilter("");
+    window.location.assign("/");
```
Comment (miguelpdiaz8) [inline frontend/src/containers/Home.tsx]: Same here.

File: frontend/src/containers/Home.tsx (hunk 1 above)
Comment (kharbutli-meta) [reply frontend/src/containers/Home.tsx]: Done.

File: frontend/src/containers/Home.tsx (hunk 2 above)
Comment (kharbutli-meta) [reply frontend/src/containers/Home.tsx]: Done.

**Review by miguelpdiaz8 — APPROVED** [review]

**Pull request merged**